### DISCUSSION TITLE
Boot Loop

### DISCUSSION THREAD

**Discussion by Oldzoot** [thehighnibble/thehighnibble.github.io#160] [IMSAI8080esp replica]
Greetings!    I have built my replica up to the point where switches are soldered in.  The system now seems to be caught in a loop as follows:

ets Jun  8 2016 00:22:57

rst:0xc (SW_CPU_RESET),boot:0x3f (SPI_FAST_FLASH_BOOT)
configsip: 188777542, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:1
load:0x3fff0018,len:4
load:0x3fff001c,len:4492
load:0x40078000,len:13324
ho 0 tail 12 room 4
load:0x40080400,len:3796
entry 0x40080624
I (354) psram: This chip is ESP32-PICO
I (354) spiram: Found 64MBit SPI RAM device
I (354) spiram: SPI RAM mode: flash 80m sram 80m
I (357) spiram: PSRAM initialized, cache is in low/high (2-core) mode.
I (364) cpu_start: Pro cpu up.
I (368) cpu_start: Application information:
I (373) cpu_start: Project name:     imsaisim_esp32
I (378) cpu_start: App version:      v1.9.1
I (383) cpu_start: Compile time:     Sep  8 2021 15:28:40
I (389) cpu_start: ELF file SHA256:  8a0e39da91675446...
I (395) cpu_start: ESP-IDF:          v4.0.2-4-g877a28e05
I (401) cpu_start: Starting app cpu, entry point is 0x40081b58
I (392) cpu_start: App cpu up.
I (898) spiram: SPI SRAM memory test OK
I (962) heap_init: Initializing. RAM available for dynamic allocation:
I (962) heap_init: At 3FFAE6E0 len 00001920 (6 KiB): DRAM
I (964) heap_init: At 3FFC36F8 len 0001C908 (114 KiB): DRAM
I (970) heap_init: At 3FFE0440 len 00003AE0 (14 KiB): D/IRAM
I (976) heap_init: At 3FFE4350 len 0001BCB0 (111 KiB): D/IRAM
I (983) heap_init: At 4009D62C len 000029D4 (10 KiB): IRAM
I (989) cpu_start: Pro cpu start user code
I (994) spiram: Adding pool of 3241K of external SPI memory to heap allocator
I (1014) spi_flash: detected chip: gd
I (1014) spi_flash: flash io: qio
I (1015) cpu_start: Starting scheduler on PRO CPU.
I (0) cpu_start: Starting scheduler on APP CPU.
I (1084) nvs: Reading settings from NVS
I (1085) nvs: settings = 00000000
I (1085) esp32_boot: Log Level set to NONE
I (1726) phy: phy_version: 4390, 6b3c1f2, Sep 10 2020, 15:09:07, 0, 0

I think line 357 means the psram is happy

I do see imsai8080 wifi for a bit,  then it disappears.  I have not been able to connect to it - it seemed to disappear after a connection attempt.

Thanks in advance for any help.
Jim

**Comment by Oldzoot** [comment]
As a test,  I started the system without the flash memory card,  there was no console output.

**Comment by nbreeden2** [comment]
Yes, it looks like your PSRAM is ok.
Have you reviewed the info on the High Nibbles site: https://thehighnibble.com/imsai8080/configure/#getting-connected
When you connect your PC to the IMSAI's AP (Access Point) are you being prompted for the IMSAIs WiFi password (this isn't your WiFi password).
When you say "it disappears" do you mean you stop seeing it in the WiFi manager on your computer? If so, you may have a security setting on your PC not allowing the connection.
Have you tried to connect to the WebUI after it disappears. The URLs are provided in the document I pointed at above. It differs depending on your platform.

**Comment by Oldzoot** [comment]
I had swapped the MAX 232 chip with another chip used in the system.   The perils of trying to build something while living an interrupt-driven life!    Fortunately, the magic smoke stayed where it belonged and now everything works! 
I believe that the AP is only available after being initialized.  The boot loop would initialize it, it would appear on my computer but before the connection was fully established the chip would restart ( possibly a signal from the ESP going to a chip and coming back to somehow confuse or reset the ESP ? ) and take the AP down.
